### PR TITLE
Editing details panel behavior

### PR DESCRIPTION
**Pull request by nunibaranes**
- support to update the outline in real-time when the user editing an item.
- improved isValid methods with trim().

### CODE REVIEW

File: src/components/Screen/CourseEditorScreen/QuizEditForm/Answers.tsx [hunk 1]
```diff
@@ -25,37 +26,47 @@ export default function Answers({
...
+          placeholder="Text"
+          value={answer.text}
+          errorMessage={fieldErrorMessage(answer.text)}
+          onChange={(e) => {
+            debouncedOnChange(index, e.target.value);
+          }}
         />
-      )}
-    </div>
+        {Boolean(answers.length > 2) && (
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/QuizEditForm/Answers.tsx:52]: `answers.length > 2` is already a boolean.

File: src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx [hunk 2]
```diff
@@ -1,25 +1,35 @@
 import React, { ReactElement, ReactNode } from 'react';
 import { Button, QuizScreen } from '@walkme/types';
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx:2]: Please remove `QuizScreen` as it no longer used.

File: src/components/Screen/CourseEditorScreen/QuizEditForm/QuestionScreenForm.tsx [hunk 3]
```diff
@@ -34,15 +39,24 @@ export default function QuestionScreenForm({ question }: { question: QuizQuestio
...
 
+  const onChange = useCallback(
+    ({ prop, value }: { prop: string; value: any }) => {
+      question[prop] = value;
+      dispatch({ type: ActionType.UpdateCourseOutline, updateHasChange: true });
+    },
+    [dispatch, question],
+  );
+
+  const debouncedOnChange = useDebounceCallback(onChange, 400);
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/QuizEditForm/QuestionScreenForm.tsx:50]: 🚀

File: src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx [hunk 4]
```diff
@@ -1,25 +1,35 @@
...
+
 import classes from './style.module.scss';
 
 export default function QuizScreenForm({
   screen,
   isWelcomeScreen,
   renderExtra,
 }: {
-  screen: QuizScreen;
+  screen: IQuizScreen; // IQuizScreen is indexable; not a new property
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx:20]: This comment seems unclear to me here. Consider moving it.

**Review by dvira — COMMENTED** [review]
Looks great! 🔥🔥🔥

File: src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx (hunk 4 above)
Comment (nunibaranes) [reply src/components/Screen/CourseEditorScreen/QuizEditForm/QuizScreenForm.tsx]: Please look at my last changes :)

File: src/components/Screen/CourseEditorScreen/QuizEditForm/interface.ts [hunk 5]
```diff
@@ -1,6 +1,18 @@
...
+
 export enum QuizScreenType {
   WelcomeScreen = 'welcome-screen',
   FailScreen = 'fail-screen',
   SuccessScreen = 'success-screen',
   QuestionScreen = 'question-screen',
 }
+
+// indexable types
+// https://levelup.gitconnected.com/introduction-to-typescript-interfaces-indexable-types-d66958523518
```
Comment (dvira) [inline src/components/Screen/CourseEditorScreen/QuizEditForm/interface.ts:11]: 💪🏽💪🏽

**Review by dvira — APPROVED** [review]
AMAZING JOB!

Approved!

**Pull request merged**